Module3_2.py

```diff
@@ -1 +1,85 @@
 
+
+class Product:
+    """This class describes a product"""
+
+    def __init__ (self, name: str, price: float) -> None:
+        self.name = name
+        self.price = price
+
+    def calc_product_total(self, quantity):
+        return float(self.price*quantity)
+
+    def __eq__(self, other):
+        if self.name == other.name and self.price == other.price:
+            return True
+        return False
+
+
+class ShoppingCart():
+    """This class describes a ShoppingCart
+
+    Attributes
+    __________
+    cart : list
+    quantity_cart: list
+
+    Methods
+    __________
+    add_to_cart(self, product: Product, quantity: int)
+        Adds a product to the cart and adds the quantity to the quantity_cart
+
+    calc_total(self)
+        Calculates the total price of all products added to the cart
+    """
+    def __init__(self):
+        self.cart = []
+        self.quantity_cart = []
+
+    def add_to_cart(self, product: Product, quantity: int):
+        """Adds  a product to the cart and adds the quantity to the quantity_cart."""
+        if product in self.cart:
+            index_product = self.cart.index(product)
+            self.quantity_cart[index_product] += quantity
+        else:
+            self.cart.append(product)
+            self.quantity_cart.append(quantity)
+
+
+    def calc_total(self):
+        """Calculates the total price of all products added to the cart."""
+
+        total = 0
+        for p in range(len(self.cart)):
+            total += self.cart[p].calc_product_total(self.quantity_cart[p])
+        return total
+
+    def __add__(self, other):
+        """Combines two carts into one"""
+
+        joined_cart = ShoppingCart()
+        joined_cart.cart += self.cart
+        joined_cart.quantity_cart += self.quantity_cart
+
+        for product, quantity in zip(other.cart, other.quantity_cart):
+            joined_cart.add_to_cart(product, quantity)
+            return joined_cart
+
```

Comment:
А тут перевіряти, якщо додаєть товар
```suggestion
if isinstance(other, Product):
    joined_cart.add_to_cart(other, 1)
```
За умовами завдання.

Reply:
Додала перевірку в метод add_to_cart

Reply:
Так вона ж там і раніше була. Ні?
<img width="866" alt="Screenshot 2022-12-08 at 13 05 04" src="https://user-images.githubusercontent.com/22391544/206431170-14e37d6a-bf39-4416-8b19-d0b03d2b97b4.png">

Проблема у тому, що у тебе не вийде додати до кошика один товар через `+`. А задання саме у тому, щоб можна було використовувати:
```python
cart = cart_obj1 + cart_obj2
cart = cart_obj + product_obj
```